px4_ros2_cpp/src/control/setpoint_types/experimental/trajectory.cpp

```diff
@@ -42,6 +42,12 @@ void TrajectorySetpointType::update(
 
 }
 
+void TrajectorySetpointType::update(const px4_msgs::msg::TrajectorySetpoint& setpoint)
```

Comment:
Which parts are you missing from the above interface? Position? jerk?

Reply:
Yes I want set the entire message directly

Reply:
Can you rather add another `update` method that allows for that? A method is just easier to use, and makes it clearer which fields are optional.

Reply:
IMO a method with 6 arguments is harder to use than a method with a single struct. The only advantage I can see for using a method with std::optional is to force the implementation to provide nullopt which ensures those values are NAN, rather than the user needing to remember to set those fields to NAN before passing the struct. Is that the reason you prefer it?

Reply:
Yes and a few other factors (the message contains other fields that do not need to be set (timestamp), having to convert from eigen vectors and API consistency).
For the too-many-arguments case I'd rather introduce an intermediate struct that allows to use Eigen vectors and std::optional.